Decred Wallet/Utils/DecredAddressURI.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  DecredAddressURI.swift
+//  Decred Wallet
+//
+// Copyright (c) 2018-2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+import Foundation
+
+struct DecredAddressURI {
+    var amount: String?
+    var address: String?
+    var memo: String?
+    
+    init(uriString: String) {
+        if uriString.contains("?amount") || uriString.contains("memo") {
+            let allParams = uriString.components(separatedBy: "?")
+            self.address = allParams[0].replacingOccurrences(of: "decred:", with: "")
+            
+            if allParams[1].contains("&") {
+                let params = allParams[1].components(separatedBy: "&")
+                let amountString = params.filter { $0.contains("amount=") }.first
+                let memoString = params.filter { $0.contains("memo") }.first
+                self.amount = amountString!.replacingOccurrences(of: "amount=", with: "") ?? ""
+                self.memo = memoString!.replacingOccurrences(of: "memo=", with: "") ?? ""
+            } else {
+                self.amount =  allParams[1].replacingOccurrences(of: "amount=", with: "")
+            }
+        } else {
+            self.address = uriString.replacingOccurrences(of: "decred:", with: "")
+        }
+    }
+}
```

Comment:
I would implement this part like this (inspired by the android code heavily):

```
	init(uriString: String) {
        let schemeSeparatedParts = uriString.components(separatedBy: ":")
        if (schemeSeparatedParts.count == 2 && schemeSeparatedParts[0] == "decred") {
            let addressAndQuery = schemeSeparatedParts[1].components(separatedBy: "?")
            address = addressAndQuery[0]

            let queryParameters = addressAndQuery[1].components(separatedBy: "&")
            for query in queryParameters {
                let nameValuePair = query.components(separatedBy: "=")
                let queryName = nameValuePair[0]
                if (nameValuePair.count == 2 && queryName == "amount") {
                    amount = Double(nameValuePair[1].trimmingCharacters(in: .whitespacesAndNewlines))
                }
            }
        }
    }

Reply:
this will totally cause a crash where another label is not present and the string does not contain an `&`. the current implementation accounts for cases where a label and I will not be implementing this.